### PR TITLE
Modify RequestConfig Javadoc

### PR DESCRIPTION
**Pull request by bekker**
Cleaned up duplicate sentences and removed 'system default' from getSocketTimeout comment ([HTTPCLIENT-1765](https://issues.apache.org/jira/browse/HTTPCLIENT-1765)).

### CODE REVIEW

File: httpclient/src/main/java/org/apache/http/client/config/RequestConfig.java [hunk 1]
```diff
@@ -295,11 +293,11 @@ public int getConnectTimeout() {
 
     /**
      * Defines the socket timeout ({@code SO_TIMEOUT}) in milliseconds,
-     * which is the timeout for waiting for data  or, put differently,
+     * which is the timeout for waiting for data or, put differently,
      * a maximum period inactivity between two consecutive data packets).
      * <p>
      * A timeout value of zero is interpreted as an infinite timeout.
-     * A negative value is interpreted as undefined (system default).
+     * A negative value is interpreted as undefined.
```
Comment (ok2c) [inline httpclient/src/main/java/org/apache/http/client/config/RequestConfig.java]: @bekker Thank you for cleaning up the javadocs but why do you want to remove `(system default)` from socket timeout description and leave it everywhere else?

Comment (bekker) [reply httpclient/src/main/java/org/apache/http/client/config/RequestConfig.java]: I was not so sure about other timeouts. Would it be better to make it same for other timeouts?

Comment (garydgregory) [reply httpclient/src/main/java/org/apache/http/client/config/RequestConfig.java]: I think we should keep the docs as is or expand on it, as opposed to removing information.

Comment (bekker) [reply httpclient/src/main/java/org/apache/http/client/config/RequestConfig.java]: @garydgregory You're right. I should add more info, not removing it.

But I'm not sure about meaning of 'undefined' here - is it 'undefined behavior'? Then there should be a warning about unexpected results.

**Comment by bekker** [comment]
OK, it seems inappropriate to fix timeout comments right now.

I reverted timeout default comments and only javadoc cleanup left.

**Comment by ok2c** [comment]
@bekker We could phrase it like ` A negative value is interpreted as undefined (system default if applicable)` if that makes it sound better to you.
Could you please also squash your commits into a single one?

**Pull request merged**